[FE] feat: Add file extension to language mapping for Monaco Editor

pigpgw opened the pull request:
## 주요 변경사항
- 파일 확장자 기반 언어 감지 기능 구현
- 파일 타입별 모나코 에디터 언어 자동 설정

## 리뷰어에게...

## 관련 이슈

closes

pigpgw committed:
feat: Add file extension to language mapping for Monaco Editor

- Add language mapping dictionary for file extensions
- Create helper function to detect language from filename
- Support all Monaco Editor's officially supported languages
- Implement automatic language detection for syntax highlighting

Example:
- .ts/.tsx -> typescript
- .js/.jsx -> javascript
- .py -> python
etc.

pigpgw committed:
Merge branch 'main' of https://github.com/jungle-6-3/code-sync-fe into feat/5

loopy-lim commented on src/lib/file.ts:
ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

loopy-lim commented on src/stores/github.ts:
```diff
@@ -106,6 +108,7 @@ export const prInfoStore = create<PrInfoPropsStore>()((set) => ({
         return {
           filename: commit.filename,
           status: commit.status,
+          language: getLanguageFromFileName(commit.filename.split(".")[1]),
```
맨 마지막이어야 해서, 1보다는 `.at(-1)`이 더 좋아요
```suggestion
          language: getLanguageFromFileName(commit.filename.split(".").at(-1),
```

pigpgw replied:
오호 이런 방법이 감사합니다. 바로 반영하도록 하겠습니다!

pigpgw committed:
refactor: Improve file extension extraction logic

- Replace split()[1] with at(-1) method for extension extraction
- Enhance reliability of last segment extraction in filenames
